torch/_higher_order_ops/triton_kernel_wrap.py

```diff
@@ -179,7 +249,7 @@ def triton_kernel_wrapper_functional_dense(
         for key, val in kwargs.items()
     }
     triton_kernel_wrapper_mutation(kernel_idx=kernel_idx, grid=grid, kwargs=kwargs)
-    return kwargs
+    return {key: val for key, val in kwargs.items() if key in tensors_to_clone}
```

Comment:
Why did we return all kwargs before and now only cloned tensor kwargs? Is it because now we effectively don't know the real outputs and don't care about returning the rest?

Reply:
Previously everything was cloned but not some arguments are input only. This is required for autograd to work correctly